R/character.R

```diff
@@ -81,11 +79,16 @@ explain.character <- function(x, explainer, labels = NULL, n_labels = NULL,
   assert_that(is.count(n_permutations))
 
   case_perm <- permute_cases(x, n_permutations, explainer$tokenization, explainer$keep_word_position)
-  case_res <- explainer$preprocess(case_perm$permutations) %>%
-    predict_model(x = explainer$model, newdata = ., type = o_type)
+  permutations_tokenized <- explainer$preprocess(case_perm$permutations)
+  case_res <- predict_model(x = explainer$model, newdata = permutations_tokenized, type = o_type)
   assert_that(length(case_perm$permutations) == n_permutations * length(x))
-  case_ind <- length(x) %>% seq() %>% map(~ rep(.x, n_permutations)) %>% flatten_int() %>% split(seq_along(case_perm$permutations), .)
-
+  
+  case_ind <- local({
+    case_range <- seq(x)
+    case_ids <- unlist(lapply(case_range, rep, n_permutations))
+    split(seq_along(case_perm$permutations), case_ids)
+  })
+  
```

Comment:
Is the use of `local` simply to avoid polluting the function namespace? If so I think we can safely remove it...

Reply:
Now it s more about readability for myself, future myself, you, future contributors,...., it makes obvious that variables created in local() are temp, just here to make things clear, and are never used elsewhere, it s my kind of workaround when I don't write with magrittr :-)

Reply:
Fair enough - I never use it myself (creates proper functions instead), but if you prefer this style I have no problem with it

Reply:
Local is just an ugly way to get an anonymous function, if { } had an effect on the scope I would use it.